lib/utils/report-deprecated-properties.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const _ = require('lodash');
+const logDeprecation = require('./logDeprecation');
+
+module.exports = (code, props, { serviceConfig } = {}) => {
+  const msg = module.exports.getMessage(props, serviceConfig);
+  if (msg) {
+    logDeprecation(code, msg, { serviceConfig });
+  }
+};
+
+module.exports.getMessage = (props, serviceConfig) => {
```

Comment:
This would work, however logDeprecation is messing around with line breaks/indentation and injecting colors. This makes this test reliant on business logic inside the `logDeprecation`. 

```
"Serverless: \u001b[38;2;255;165;0mDeprecation warning: Starting with version 3.0.0, following properties will be replaced:\u001b[39m\n\u001b[38;2;255;165;0m              \"provider.role\" -> \"provider.iam.role\"\u001b[39m\n\u001b[38;2;255;165;0m              \"provider.iamRoleStatements\" -> \"provider.iam.role.statements\"\u001b[39m\n            \u001b[38;2;255;165;0mMore Info: https://www.serverless.com/framework/docs/deprecations/#MY_CODE\u001b[39m\n"
```

vs 

```
Starting with version 3.0.0, following properties will be replaced:\n
  "provider.role" -> "provider.iam.role"\n
  "provider.iamRoleStatements" -> "provider.iam.role.statements"
```

I can make this work, however not sure how it makes it any better.

Reply:
I think confirming just on some bits as `"provider.role" -> "provider.iam.role"` will still work. So I would do just that (let's do just sanity check that message went through, but not necessarily confirm on all its characters)

Reply:
It'll work, but the test by itself will be covering only parts of what code is going. Up to you really, but I wouldn't do it this way. 

I see no harm in exposing a fn if it makes testing very specific and isolated from upstream dependencies (aka low maintenance).

Reply:
> It'll work, but the test will be covering only parts of what the code is doing

In my understanding we still can maintain 100% coverage (if we want to), and confirming on every character on string, doesn't really expand the coverage, it just enforces us to put same not simple string in two different place (which maintenance wise is problematic)

Reply:
See again please @medikoo